gateway/mw_request_signing.go

```diff
@@ -101,7 +101,12 @@ func (s *RequestSigning) ProcessRequest(w http.ResponseWriter, r *http.Request,
 	}
 
 	headers := generateHeaderList(r, s.Spec.RequestSigning.HeaderList)
-	signatureString, err := generateHMACSignatureStringFromRequest(r, headers)
+
+	path := r.URL.Path
+	if s.Spec.Proxy.StripListenPath {
+		path = s.Spec.StripListenPath(r, r.URL.Path)
+	}
+	signatureString, err := generateHMACSignatureStringFromRequest(r, headers, path)
```

Comment:
Do not think it is only about stripping listen path. What if user used URL rewrite, and updated the path, will your change pick this up? Pls test.

Reply:
Will check